pkg/collector/corechecks/snmp/internal/report/interface_bandwidth_usage_rate_test.go

```diff
@@ -0,0 +1,301 @@
+							Value: 5000000.0,
+						},
+					},
+					// ifHCOutOctets
+					"1.3.6.1.2.1.31.1.1.1.10": map[string]valuestore.ResultValue{
+						"9": {
+							Value: 1000000.0,
+						},
+					},
+					// ifHighSpeed
+					"1.3.6.1.2.1.31.1.1.1.15": map[string]valuestore.ResultValue{
+						"9": {
+							Value: 80.0,
+						},
+					},
+				},
+			},
+			// current usage value @ ts 30 for ifBandwidthInUsage
+			// ((5000000 * 8) / (80 * 1000000)) * 100 = 50.0
+			// previous usage value @ ts 15
+			// ((3000000 * 8) / (80 * 1000000)) * 100 = 30.0
+			// rate generated between ts 15 and 30
+			// (50 - 30) / (30 - 15)
+			expectedRate: 20.0 / 15.0,
+			usageValue:   50,
+		},
+		{
+			name:      "snmp.ifBandwidthOutUsage.Rate ifHCOutOctets submitted",
+			symbols:   []profiledefinition.SymbolConfig{{OID: "1.3.6.1.2.1.31.1.1.1.10", Name: "ifHCOutOctets"}},
+			fullIndex: "9",
+			values: &valuestore.ResultValueStore{
+				ColumnValues: valuestore.ColumnResultValuesType{
+					// ifHCInOctets
+					"1.3.6.1.2.1.31.1.1.1.6": map[string]valuestore.ResultValue{
+						"9": {
+							Value: 5000000.0,
+						},
+					},
+					// ifHCOutOctets
+					"1.3.6.1.2.1.31.1.1.1.10": map[string]valuestore.ResultValue{
+						"9": {
+							Value: 1000000.0,
+						},
+					},
+					// ifHighSpeed
+					"1.3.6.1.2.1.31.1.1.1.15": map[string]valuestore.ResultValue{
+						"9": {
+							Value: 80.0,
+						},
+					},
+				},
+			},
+			// current usage value @ ts 30 for ifBandwidthOutUsage
+			// ((1000000 * 8) / (80 * 1000000)) * 100 = 10.0
+			// previous usage value @ ts 15
+			// ((500000 * 8) / (80 * 1000000)) * 100 = 5.0
+			// rate generated between ts 15 and 30
+			// (10 - 5) / (30 - 15)
+			expectedRate: 5.0 / 15.0,
+			usageValue:   10,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			sender := mocksender.NewMockSender("testID") // required to initiate aggregator
+			sender.On("Gauge", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return()
+
+			TimeNow = common.MockTimeNow
+			ms := &MetricSender{
+				sender:                  sender,
+				interfaceConfigs:        tt.interfaceConfigs,
+				interfaceBandwidthState: interfaceRateMapWithPrevious(),
+			}
+
+			usageName := bandwidthMetricNameToUsage[tt.symbols[0].Name]
+			rate, err := ms.interfaceBandwidthState.calculateBandwidthUsageRate(tt.fullIndex, usageName, ifSpeed, tt.usageValue)
+			interfaceID := fullIndex + "." + usageName
+
+			// Expect no errors
+			assert.Nil(t, err)
+
+			assert.Equal(t, tt.expectedRate, rate)
+
+			// Check that the map was updated with current values for next check run
+			assert.Equal(t, ifSpeed, ms.interfaceBandwidthState[interfaceID].ifSpeed)
+			assert.Equal(t, tt.usageValue, ms.interfaceBandwidthState[interfaceID].previousSample)
+			assert.Equal(t, mockTimeNowNano, ms.interfaceBandwidthState[interfaceID].previousTsNano)
+		})
+	}
+}
+
+func Test_interfaceBandwidthState_calculateBandwidthUsageRate_logs(t *testing.T) {
```

Comment:
This isn't really testing logs, I would maybe change the name to errors or something.

Also looking at this test and the one above you could theoretically make a single table structure that would allow you to test both the non-error and error scenarios. Up to you if you want to leave them separate, that's not a blocking issue for me :)